### ISSUE TITLE
BUG: not properly closing SQLite connection

### ISSUE THREAD

**Issue by ruanlianjun** [brendan-ward/mbtiles-go#1]
### after time use conn have a pannic
```shell
/world.mbtiles garbage collected, call Close method
```
```go
// TODO: only if Debug ?
_, file, line, _ := runtime.Caller(stackCallerLayers)
runtime.SetFinalizer(conn, func(conn *Conn) {
if !conn.closed {
	var buf [20]byte
	panic(file + ":" + string(itoa(buf[:], int64(line))) + ": *sqlite.Conn for " + path + " garbage collected, call Close method")
}
})
```

**Comment by brendan-ward** [comment]
What version are you using?

Do you use a deferred `Close()` on the database after opening it?  E.g.,

```
db, err := mbtiles.Open("world.mbtiles")
if err != nil { ... }
defer db.Close()
```

Is `world.mbtiles` a valid mbtiles file or does it fail during initial validation?  If it is not valid, there was indeed a bug that was fixed in 67a7dabdbaabfe83ae008ad05bb170db035e3b1c.

If neither of those two things, can you please tell me more about how you are opening & working with `world.mbtiles` using this package?  My primary use case of this package is [mbtileserver](https://github.com/consbio/mbtileserver), so it is entirely possible that there are bugs lurking here beyond what I've used in that package, and it would be good to get to the bottom of it.

**Comment by brendan-ward** [comment]
Probably solved by #6; closing due to lack of activity
